### DISCUSSION TITLE
Extending Record Query Condition and Localization Support in Pocketbase

### DISCUSSION THREAD

**Discussion by hungcrush** [pocketbase/pocketbase#2542] [Q&A]
Hi!

Look at the roadmap. I know you have a lot of issues that need to be addressed. I absolutely love Pocketbase, and it would be amazing if it could support multiple locales. While waiting for you to have time to take care of it, I want to give it a try myself.

Here's my idea: use it as a framework.

Step 1: Add a "locale" column to every collection that supports multiple locales. When querying records, I will add an additional condition like this: "AND locale = 'current_locale'".

Step 2: Create more tables, such as "_{collection_name}_localizations_links", to handle language switching.

Now, I have a question. What's the hook for extending the record query condition? Could you provide me with some ideas to help me handle that?

Thank you so much!

**Comment by ganigeorgiev** [comment]
There is no global `SELECT` query hook. This was requested before, but I have some performance concerns and for now I put it aside. I'll consider it during the planned refactoring but even then I'm not sure in your case how the `current_locale` will be passed. An alternative could be to create the `locale='current_locale'` global condition on client-side level by having a "base filter" or similar.
